src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -391,35 +371,37 @@ public void HandleDatabaseRecordChange()
             var newRecord = LoadDatabaseRecord();
 
             if (newRecord == null)
+            {
+                DropAllOutgoingConnections();
+                MyDatabaseRecord = null;
                 return;
+            }
 
-            var connectionChanged = ReplicationTopology.MyConnectionChanged(newRecord.Topology, Server.NodeTag, Database.Name);
+            var connectionChanged = ReplicationTopology.FindConnectionChanges(newRecord.Topology, Server.NodeTag, Database.Name);
             var conflictSolverChanged = MyDatabaseRecord.ConflictSolverConfig.ConflictResolutionChanged(newRecord.ConflictSolverConfig);
 
             MyDatabaseRecord = newRecord;
 
-            if (connectionChanged)
+            if (connectionChanged.nodesToRemove.Count > 0)
             {
+                // remove old connections
                 if (_log.IsInfoEnabled)
-                    _log.Info("Replication connection were changed. Starting and stopping outgoing replication threads.");
+                    _log.Info("Stopping obselete outgoing replication threads.");
 
-                //prevent reconnecting to a destination that we shouldn't in case we have flaky network
-                _reconnectQueue.Clear();
-
-                foreach (var instance in _outgoing)
-                {
-                    instance.Failed -= OnOutgoingSendingFailed;
-                    instance.SuccessfulTwoWaysCommunication -= OnOutgoingSendingSucceeded;
-                    instance.Dispose();
-                }
+                DropOutgoingConnections(connectionChanged.nodesToRemove);
+            }
 
-                _outgoing.Clear();
-                _outgoingFailureInfo.Clear();
-                _lastSendEtagPerDestination.Clear();
+            if (connectionChanged.nodesToAdd.Count > 0)
+            {
+                // add new connections
+                if (_log.IsInfoEnabled)
+                    _log.Info("Starting new replication threads.");
```

Comment:
Same here, tell the logger what you are actually doing, so you'll have that information in the log file if needed